samples-tests/src/test/kotlin/org/gradle/kotlin/dsl/samples/MavenPublishSampleTest.kt

```diff
@@ -0,0 +1,39 @@
+package org.gradle.kotlin.dsl.samples
+
+import org.junit.Assert
+import org.junit.Test
+import java.nio.file.Paths
+import java.util.jar.JarFile
+
+class MavenPublishSampleTest : AbstractSampleTest("maven-publish") {
+    @Test
+    fun `publish`() {
+        val projectName = "maven-publish"
+        val projectVersion = "1.0.0"
+
+        // when:
+        build("publish")
```

Comment:
The test should avoid messing up with `~/.m2` by setting `maven.repo.local` as it's done in https://github.com/gradle/kotlin-dsl/blob/dcd84adf60591726b276fb644c6795dab86d85db/samples-tests/src/test/kotlin/org/gradle/kotlin/dsl/samples/MavenPluginSampleTest.kt#L13

Reply:
Done.  I'm trying to understand the concern though.  Are there instances when `publish` pushes artifacts to ~/.m2?  I thought that only happened with `publishToMavenLocal`.

Reply:
That is a pretty good question. I'd love to know the answer too.

Reply:
You're right @shradej1, `publish` does not trigger `publishToMavenLocal`.

Setting `maven.repo.local` is not necessary here. Did I miss something @bamboo?

Reply:
I assumed `publish` had similar behavior to `uploadArchives` with regards to the `maven-metadata-remote.xml` file but that doesn't seem to be the case. The property is indeed unnecessary.

Reply:
Thanks for the clarification.  Should I remove it or leave it as is?

Reply:
I just manually confirmed that `publish` does not mess with `~/.m2/`.
@shradej1 yes, please, remove it

Reply:
@eskatos reverted maven repo change and rebased.